test/SILOptimizer/side_effects.sil

```diff
@@ -1,4 +1,4 @@
-// RUN: %target-sil-opt %s -parse-serialized-sil -module-name test -compute-side-effects | %FileCheck %s
+// RUN: %target-sil-opt %s -parse-serialized-sil -module-name test -compute-side-effects -temp-lvalue-elimination | %FileCheck %s
```

Comment:
Please do not run temp-lvalue-elimination in this testfile.
Instead, just check if side effects are computed correctly in this file and add the temp-lvalue-elimination test functions to `test/SILOptimizer/templvalueopt_ossa.sil`. There are already some deinit barrier tests there, but not for callee functions which are `[readonly]`.